docs/naps/8-telemetry.md

```diff
@@ -0,0 +1,265 @@
+https://sentry.io/
+https://opentelemetry.io/
+
+Visualizations:
+https://github.com/grafana/grafana
+  
+
+  
+## Implementation 
+
+The key consideration for implementation should be the low cost of maintenance. 
+So the solution should be as simple as possible.
+We could either use existing solutions on the server side or implement our own.
+
+The benefit of existing solutions is that most of the work is already done.
+The downside is that it may require additional cost of maintenance.
+This cost may be caused by many features that are not needed for napari and could increase the risk of leaking data.
+Quick checks of their code revealed they are implemented in techniques that are not familiar to napari core devs.
+So, if we decide to use them, we should select an SAS solution that will be maintained by the company.
+
+
+For now, I suggest creating a simple REST API server for collecting the data. 
+It could be a simple Python FastAPI server that will store data in the SQLite database.
+Connection to server will be encrypted using HTTPS and certificate provided by LetsEncrypt.
+
+Data for aggregation should be extracted from the database using a script running on the same machine.
+
+The output of the aggregation script should be loaded to some existing visualization tool, like grafana.
+
+It may be nice to host raw and aggregate data on separate servers —
+then even if the data presented on the dashboard is compromised,
+the raw data will be not exposed to the world.
+
+Having both server and aggregation scripts in Python will reduce maintenance costs for napari core devs.
+
+We should register the `telemetry.napari.org` domain and use it for the server. 
+The main page will contain this NAP and a link to the summary dashboard.
+
+
+The main part of the application side should be implemented in `napari-telemetry` package. 
+The package should not report in stream mode, but collect data on the disk and send it in batches.
+This will reduce the risk of leaking data.
+The package should implement a utility to allow users to preview collected data before sending it to the server.
+
+In napari itself, the following changes should be implemented:
+
+1) The indicator that shows the telemetry status
+2) The dialog that asks a user if they want to enable telemetry
+3) code to check if telemetry is enabled (to not load the `napari-telemetry` package if it is disabled)
+4) code required to init `napari-telemetry` package
+
+
+## Potential problems 
+
+There is a risk
+that someone may try to high-jack the telemetry module name to have code executed at every napari start. 
+
+I do not expect that it is a high risk, but exists.
+We could address it by code signing.
+This will require additional procedures to protect private cryptographic keys.
+
+Another option is to scan public plugins and their dependencies. 
+This is simpler,
+but will require establishing additional communication channels to be able to warn users about the potential problem. 
```

Comment:
I started thinking that this section is obsolete and could be removed from NAP.

Reply:
@Czaki agreed, I don't think this is necessary.